Updates Ongoing Activity codelab

garanj opened the pull request:
- Migrates from views to Compose
- Adds permission logic for Android 14 POST_NOTIFICATIONS
- Fixes animated icon on watch face
- Updates dependencies and structure
- Adds spotless

ithinkihaveacat commented on finished/src/main/java/com/android/example/wear/ongoingactivity/presentation/OngoingActivityExampleApp.kt:
```diff
@@ -0,0 +1,66 @@
+
+@OptIn(ExperimentalPermissionsApi::class)
+@Composable
+fun OngoingActivityExampleApp(
```
Not sure how much you want to fiddle with this class but I think there's a few things that could be improved (some overlap between these options):

1. Handle the user rejecting the POST_NOTIFICATIONS permission. (I guess it should still work, just not display OA.)
2. Switch on Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU instead of creating a nullable `permissionState`, and then switching on that. (Would be nicer if we could either rememberPermissionState() with an empty list of permission *or* construct our own PermissionState object with PermissionState.Granted but neither seems to be possible…)
3. Eliminate the duplicate Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU test. (Perhaps just check if permissionState is null?)
4. Implement/support shouldShowRequestPermissionRationale() (especially since the generic notification permission message doesn't explain that it's required to get the OA indicator).

ithinkihaveacat reviewed (commented):
Not an approver but LGTM.

Great improvement on the old version that existed before!

shumelchyk reviewed (approved):
Thanks Garan, LGTM!
(and great questions from @ithinkihaveacat)

garanj commented:
Thanks @ithinkihaveacat !

I've addressed some of these issues I think, but it could do with verification, particularly on (4):

2. (and 3) Changed only to one version check and created a `PermissionState` set to `status` = `GRANTED` for < T.
4. Added use of `shouldShowRationale` - this is slightly fiddly as this can be `false` in two scenarios:
    1. When the permission has never been requested before
    2. When the rationale has been shown before (but permission rejected)
    For this reason, it's necessary to keep track of whether the rationale has been shown before, to distinguish these two cases.

I didn't add (1) as that potentially suggests that using the OA is optional, whereas I think we want to be requiring it (i.e. as per Play guidelines) where it makes sense.

ithinkihaveacat commented:
Looks great, LGTM!

(I looked over the code in the `finished` directory, I didn't check to see how it integrates with the codelab published at https://developer.android.com/codelabs/ongoing-activity.)
